File: remove_bg.py
import sys
from PIL import Image

def remove_checkerboard(img_path, out_path):
    img = Image.open(img_path).convert("RGBA")
    data = img.load()
    width, height = img.size
    
    # Get the two colors of the checkerboard from the top-left corner
    # The checkerboard is usually a grid, so we can sample (0,0) and say (10,10) or find the two dominant colors in the background
    # Actually, we can just look at the top left 32x32 pixels which should be pure background
    bg_colors = set()
    for x in range(32):
        for y in range(32):
            r, g, b, a = data[x, y]
            if a > 0:
                bg_colors.add((r, g, b))
                
    print("Found background colors:", bg_colors)
    
    # Tolerance for color matching
    tol = 15
    
    def match(c1, c2):
        return abs(c1[0]-c2[0]) < tol and abs(c1[1]-c2[1]) < tol and abs(c1[2]-c2[2]) < tol

    for y in range(height):
        for x in range(width):
            r, g, b, a = data[x, y]
            is_bg = False
            for bc in bg_colors:
                if match((r, g, b), bc):
                    is_bg = True
                    break
            if is_bg:
                data[x, y] = (0, 0, 0, 0)
                
    img.save(out_path)
    print("Saved transparent image to", out_path)

if __name__ == "__main__":
    remove_checkerboard(sys.argv[1], sys.argv[2])

File: remove_bg.js
const { Jimp } = require("jimp");

async function removeBackground() {
    const imagePath = "src/assets/burning_pixel_card.png";
    console.log("Loading image...", imagePath);
    const image = await Jimp.read(imagePath);
    
    // Sample a few pixels in the top-left corner to get the checkerboard colors
    const bgColors = [];
    for (let x = 0; x < 32; x++) {
        for (let y = 0; y < 32; y++) {
            const hex = image.getPixelColor(x, y);
            const r = (hex >> 24) & 255;
            const g = (hex >> 16) & 255;
            const b = (hex >> 8) & 255;
            const colorStr = `${r},${g},${b}`;
            if (!bgColors.includes(colorStr)) {
                bgColors.push(colorStr);
            }
        }
    }
    
    console.log("Found background colors:", bgColors);
    
    const tolerance = 25;
    
    image.scan(0, 0, image.bitmap.width, image.bitmap.height, function (x, y, idx) {
        const r = this.bitmap.data[idx + 0];
        const g = this.bitmap.data[idx + 1];
        const b = this.bitmap.data[idx + 2];
        
        let isBg = false;
        for (const bc of bgColors) {
            const [bcr, bcg, bcb] = bc.split(",").map(Number);
            if (Math.abs(r - bcr) < tolerance && Math.abs(g - bcg) < tolerance && Math.abs(b - bcb) < tolerance) {
                isBg = true;
                break;
            }
        }
        
        if (isBg) {
            this.bitmap.data[idx + 3] = 0; // alpha to 0
        }
    });
    
    await image.write(imagePath);
    console.log("Done!");
}

removeBackground().catch(console.error);
